### ISSUE TITLE
ExternalRefprocessor fails to load $ref properties from ArraySchema with inline items object

### ISSUE THREAD

**Issue by scottr-ad** [swagger-api/swagger-parser#1170]
For schemas located in a externally referenced file, if an Array schema has items as an inline Object, any $ref properties fail to load. However if the Array schema specifies its items as a $ref, the sub-refs load fine.

swagger.yaml
---------------
```
openapi: "3.0.2"
info:
  version: 15.3.0
  title: test
paths:
  /add-breeds:
    put:
      requestBody:
        content:
          application/json:
            schema:
              $ref: 'common.yaml#/components/schemas/BreedsList'
      responses:
        '200':
          description: Expected response to a valid request
          content:
            text/plain:
              schema:
                type: string

  /add-pets:
    put:
      requestBody:
        content:
          application/json:
            schema:
              $ref: 'common.yaml#/components/schemas/PetsList'
      responses:
        '200':
          description: Expected response to a valid request
          content:
            text/plain:
              schema:
                type: string

```

common.yaml
----------------
```
openapi: "3.0.2"
info:
  version: 15.3.0
  title: "Common Data Types"
paths: {}
components:
  schemas:
    BreedsList:
      type: array
      items:
        $ref: '#/components/schemas/BreedsListItems'

    BreedsListItems:
      type: object
      properties:
        breed:
          $ref: '#/components/schemas/Breed'

    Breed:
      type: object
      properties:
        name:
          type: string
        family:
          type: string

    PetsList:
      type: array
      items:
        type: object
        properties:
          petType:
            type: string
          colouring:
            $ref: '#/components/schemas/Colouring'

    Colouring:
      type: object
      properties:
        primary:
          $ref: '#/components/schemas/Colour'
        secondary:
          $ref: '#/components/schemas/Colour'

    Colour:
      type: string
      enum: ['black', 'white', 'tan', 'red', 'blue']

```

In the example above, $refs to BreedsList, BreedsListItem and Breed all are loaded, however only $refs to PetsList are loaded (not Colouring or Colour)

This is due to the following line in ExternrefProcessor, which only drills into ArraySchema if its items is a $ref, and misses inline object items with sub-properties.

ExternalRefProcessor
-----------------------
```
            if (schema instanceof ArraySchema && ((ArraySchema) schema).getItems() != null && ((ArraySchema) schema).getItems().get$ref() != null
                    && StringUtils.isNotBlank(((ArraySchema) schema).getItems().get$ref())) {
                processRefSchema(((ArraySchema) schema).getItems(), file);
            }

```

The solution is to modify this section to process item properties when $ref is blank.

**Comment by scottr-ad** [comment]
Pull Request #1171 raised to fix this issue. Could one of the regular contributors please review and approve or provide feedback?

**Comment by gracekarina** [comment]
closed by #1171
